### PR TITLE
feat(new-rule): summary elements must have an accessible name

### PR DESCRIPTION
**Pull request by WilcoFiers**
This rule checks that summary elements have an accessible name, through text content, aria-label(ledby) or title. It skips summary elements that are not used as controls for `details`, or if its `details` element has no content.

Closes: #4510

### CODE REVIEW

File: lib/rules/summary-interactive-matches.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  const firstSummary = parent.children.find(
+    child => child.props.nodeName === 'summary'
+  );
+  if (firstSummary !== virtualNode) {
+    return false;
+  }
+  return true;
+}
+
+function hasDetailsContent(vDetails) {
```
Comment (straker) [inline lib/rules/summary-interactive-matches.js:25]: Why ignore summary elements of details that don't have content? They are still interactive and screen readers read them out. They don't seem to be treated any differently if the details has text content.

```html
<details>
  <summary></summary>
</details>
```

* JAWS / Chrome - "unlabeled zero button, collapsed"
* NVDA / Firefox - "filled right pointing small triangle, collapsed"
* VO / Safari - "collapsed, details"

**Review by straker — CHANGES_REQUESTED** [review]

File: lib/rules/summary-interactive-matches.js (hunk 1 above)
Comment (WilcoFiers) [reply lib/rules/summary-interactive-matches.js:25]: I'm not less sure what's going on when there's nothing in the details element to expand. Its just an empty tab stop at that point. Those aren't great, but they're not an WCAG violation IMO.

File: lib/rules/summary-interactive-matches.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { hasContentVirtual } from '../commons/dom';
+
+export default function summaryIsInteractiveMatches(_, virtualNode) {
+  // Summary only interactive if its real DOM parent is a details element
+  const parent = virtualNode.parent;
+  if (
+    parent.props.nodeName !== 'details' ||
+    virtualNode.shadowId !== parent.shadowId
```
[the remark this answers is not in the document]
Comment (WilcoFiers) [reply lib/rules/summary-interactive-matches.js:8]: Gd catch. Do have a way to see in virtual tree that two nodes are in the same tree?

[the remark this answers is not in the document]
Comment (straker) [reply lib/rules/summary-interactive-matches.js:8]: I don't know if we have a way to distinguish between light DOM tree and shadow DOM tree.

**Review by straker — APPROVED** [review]

**Pull request merged**